spec/haskell/src/SEL4/Object/Reply.lhs

```diff
@@ -155,17 +156,18 @@ This module specifies the behavior of reply objects.
 >         setReply prevReplyPtr (prevReply { replyNext = Nothing })
 
 >     cleanReply rptr
->     replyUnlink rptr
+>     replyUnlink rptr tptr
 
-> replyUnlink :: PPtr Reply -> Kernel ()
-> replyUnlink replyPtr = do
+> replyUnlink :: PPtr Reply -> PPtr TCB -> Kernel ()
+> replyUnlink replyPtr tcbPtr = do
 >     tptrOpt <- getReplyTCB replyPtr
 >     tptr <- maybeToMonad tptrOpt
->     state <- getThreadState tptr
+>     assert (tptr == tcbPtr) "replyTCB is equal to tcbPtr"
```

Comment:
Is this assertion necessary?

Reply:
It would certainly make showing `sym_refs` easier.

Reply:
If the `assert` is proof-motivated, it looks fine (it does look like it would help with `sym_refs`). If it is just there because there is one in C, we should remove it.

Oh, also the message should be about the failed assert, i.e.
```suggestion
>     assert (tptr == tcbPtr) "replyTCB is not equal to tcbPtr"
```

or

```suggestion
>     assert (tptr == tcbPtr) "replyTCB must be equal to tcbPtr"
```

Reply:
> If the `assert` is proof-motivated, it looks fine (it does look like it would help with `sym_refs`). If it is just there because there is one in C, we should remove it.

At this point, I alone can’t conclude either way. Whoever actually is going to use this soon would have better ideas, and we can take it from there. Also, I am the one who added it to the C as well at this point. 

> Oh, also the message should be about the failed assert, i.e.

Right, I was wondering about that, and got confused at some point! I’ll fix those up.